scripts/agent_registry.py

```
@@ -0,0 +1,272 @@
+    def add_training(self, training_name: str, completion_date: str, score: float = None):
+        """Add completed training to the agent's record."""
+        training_record = {
+            'training_name': training_name,
+            'completion_date': completion_date,
+            'score': score,
+            'timestamp': datetime.now().isoformat()
+        }
+        self.training_completed.append(training_record)
+        self.last_updated = datetime.now().isoformat()
+        
+    def update_performance(self, tasks_completed: int = 0, success_rate: float = None, 
```

Ruff (0.14.3)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)